FreeRTOS/FreeRTOS
# Support multiple transports in the same compilation unit

Description
-----------

By removing the definition of the NetworkContext struct in the header file, we allow the application to define it. This allows an application writer to use multiple transports in the same compilation unit. That way, multiple .c files do not have to be created for each transport.

All these changes are tested to work in CI.

Test Steps
-----------

Run the demos as you normally would.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
